tree/ntuple/v7/src/RField.cxx

```diff
@@ -916,6 +923,181 @@ void ROOT::Experimental::RClassField::AcceptVisitor(Detail::RFieldVisitor &visit
+   } else {
+      switch (fProxy->GetType()) {
+      case EDataType::kChar_t:   itemField = std::make_unique<RField<char>>("_0"); break;
+      case EDataType::kUChar_t:  itemField = std::make_unique<RField<std::uint8_t>>("_0"); break;
+      case EDataType::kShort_t:  itemField = std::make_unique<RField<std::int16_t>>("_0"); break;
+      case EDataType::kUShort_t: itemField = std::make_unique<RField<std::uint16_t>>("_0"); break;
+      case EDataType::kInt_t:    itemField = std::make_unique<RField<std::int32_t>>("_0"); break;
+      case EDataType::kUInt_t:   itemField = std::make_unique<RField<std::uint32_t>>("_0"); break;
+      case EDataType::kLong_t:
+      case EDataType::kLong64_t:
+         itemField = std::make_unique<RField<std::int64_t>>("_0");
+         break;
+      case EDataType::kULong_t:
+      case EDataType::kULong64_t:
+         itemField = std::make_unique<RField<std::uint64_t>>("_0");
+         break;
+      case EDataType::kFloat_t:  itemField = std::make_unique<RField<float>>("_0"); break;
+      case EDataType::kDouble_t: itemField = std::make_unique<RField<double>>("_0"); break;
+      case EDataType::kBool_t:   itemField = std::make_unique<RField<bool>>("_0"); break;
+      default:
+         throw RException(R__FAIL("unsupported value type"));
+      }
+   }
+   fItemSize = itemField->GetValueSize();
+   Attach(std::move(itemField));
+}
+
+std::unique_ptr<ROOT::Experimental::Detail::RFieldBase>
+ROOT::Experimental::RCollectionClassField::CloneImpl(std::string_view newName) const
+{
+   return std::unique_ptr<RCollectionClassField>(
+      new RCollectionClassField(newName, GetType(), fProxy->GetCollectionClass()));
+}
+
+std::size_t ROOT::Experimental::RCollectionClassField::AppendImpl(const Detail::RFieldValue &value)
+{
+   TVirtualCollectionProxy::TPushPop RAII(fProxy, value.GetRawPtr());
+   std::size_t nbytes = 0;
+   auto count = fProxy->Size();
+   for (unsigned i = 0; i < count; ++i) {
+      auto itemValue = fSubFields[0]->CaptureValue(fProxy->At(i));
+      nbytes += fSubFields[0]->Append(itemValue);
+   }
+   Detail::RColumnElement<ClusterSize_t> elemIndex(&fNWritten);
+   fNWritten += count;
+   fColumns[0]->Append(elemIndex);
+   return nbytes + sizeof(elemIndex);
+}
+
+void ROOT::Experimental::RCollectionClassField::ReadGlobalImpl(NTupleSize_t globalIndex, Detail::RFieldValue *value)
+{
+   TVirtualCollectionProxy::TPushPop RAII(fProxy, value->GetRawPtr());
+
+   ClusterSize_t nItems;
+   RClusterIndex collectionStart;
+   fPrincipalColumn->GetCollectionInfo(globalIndex, &collectionStart, &nItems);
+
+   auto oldNItems = fProxy->Size();
+   if (fProxy->GetProperties() & TVirtualCollectionProxy::kNeedDelete) {
+      for (std::size_t i = 0; i < oldNItems; ++i) {
+         auto itemValue = fSubFields[0]->CaptureValue(fProxy->At(i));
+         fSubFields[0]->DestroyValue(itemValue, true /* dtorOnly */);
+      }
+   }
+   fProxy->Clear();
```

Comment:
`Clear` already included the above code (See `TGenCollectionProxy::Clear`, doesn't it?

Reply:
Yes, but that's only in the derived `TGenCollectionProxy` class; we cannot rely on other implementations of `TVirtualCollectionProxy::Clear()` doing the same, no?

Reply:
Humm .. that would probably be a "bug" if they did not.  However I was slightly wrong, in order to get the deletion on the pointee to object you need:
```
fProxy->Clear("force");
```

Reply:
On second thoughts, I'm actually not convinced about replacing _all_ the code above by
```c++
fProxy->Clear("force");
```

Rationale: in RNTuple we rely on `GenerateValue()` / `DestroyValue()` to construct/destruct objects of a specific type (represented by a `RField` instance).  Given that the value type of the collection is kept as a sub-field, I think it is appropriate to let the `RField` "see" the `DestroyValue()` call.
What do you think, @jblomer?

Reply:
On the other hand, this duplicates code and prevents a collection proxy implementation to use accelerator for specific type of collections.